habit/models.py

```diff
@@ -0,0 +1,34 @@
+from django.db import models
+from datetime import datetime
+from django.contrib.auth.models import AbstractBaseUser
+# Create your models here.
+
+
+class User(AbstractBaseUser):
+    def __repr__(self):
+        return f"<User username={self.username}>"
+    
+    def __str__(self):
+        return self.username
+
+
+class Habit(models.Model):
+    name = models.CharField(max_length=50)
+    goal = models.CharField(max_length=200)
+    user = models.ForeignKey(User,on_delete=models.CASCADE,related_name="habit_user"
+    )
+
+
+class Record(models.Model):
+    date = models.DateField(auto_now_add=datetime.now)
+    habit = models.ForeignKey(
+        Habit, on_delete=models.CASCADE, null=True, blank=True, related_name="habit"
+    )
+    user = models.ForeignKey(
```

Comment:
Since you have a foreign key for User on the Habit model, you don't need to have it here as well. You will know what record belongs to what user because the Record has a FK relationship with Habit (which has one with User).